src/utils/DateUtils.ts

```diff
@@ -29,111 +130,174 @@ class DefaultDateToStringOptions extends Options implements IDateToStringOptions
+   * @returns {string} Corresponds to the Date in a 'YYYY/MM/DD' format.
+   */
+  static dateForQuery(date: Date): string {
+    DateUtils.setLocale();
+    const dateMoment = moment(date).format('YYYY/MM/DD');
+    return dateMoment;
+  }
 
-    return newDate;
+  /**
+   * Keeps only the date information, removing the time information.
+   * @param date The date to crop.
+   * @returns {Date} The cropped date, without the time information.
+   */
+  static keepOnlyDatePart(date: Date): Date {
+    DateUtils.setLocale();
+    const dateMoment = moment(date);
+    return new Date(dateMoment.year(), dateMoment.month(), dateMoment.date());
   }
 
+  /**
+   * Converts the date into a string according to the specified options.
+   * @param d The date to convert to a string.
+   * @param options Specifies the enabled options on the date.
+   * @returns {any} The date after being formatted into a string according to the specified options.
+   */
   static dateToString(d: Date, options?: IDateToStringOptions): string {
+    DateUtils.setLocale();
+
     if (Utils.isNullOrUndefined(d)) {
       return '';
     }
 
     options = new DefaultDateToStringOptions().merge(options);
 
-    var dateOnly = DateUtils.keepOnlyDatePart(d);
+    const dateOnly = moment(DateUtils.keepOnlyDatePart(d));
 
     if (options.predefinedFormat) {
-      return Globalize.format(dateOnly, options.predefinedFormat);
+      return dateOnly.format(options.predefinedFormat.replace(/yyyy/g, String.prototype.toUpperCase()));
     }
 
-    var today = DateUtils.keepOnlyDatePart(options.now);
+    const today = moment(DateUtils.keepOnlyDatePart(options.now));
     if (options.useTodayYesterdayAndTomorrow) {
-      if (dateOnly.valueOf() == today.valueOf()) {
-        return l('Today');
-      }
-      var tomorrow = DateUtils.offsetDateByDays(today, 1);
-      if (dateOnly.valueOf() == tomorrow.valueOf()) {
-        return l('Tomorrow');
-      }
-      var yesterday = DateUtils.offsetDateByDays(today, -1);
-      if (dateOnly.valueOf() == yesterday.valueOf()) {
-        return l('Yesterday');
+      /*
+       'today' is already a moment, but we need to create another instance of moment so that 'today' isn't modified by 'add'.
+       If we simply used 'today', we would have to add(-2,'days') instead of -1.
+       */
+      if (dateOnly.valueOf() == today.valueOf() || dateOnly.valueOf() == moment(today).add(1, 'days').valueOf() ||
+        dateOnly.valueOf() == moment(today).add(-1, 'days').valueOf()) {
+        return moment(dateOnly).calendar(moment(today));
       }
     }
 
-    var isThisWeek = Math.abs(TimeSpan.fromDates(dateOnly, today).getDays()) < 7;
+
+    const isThisWeek = dateOnly.diff(moment(today), 'weeks') == 0;
     if (options.useWeekdayIfThisWeek && isThisWeek) {
       if (dateOnly.valueOf() > today.valueOf()) {
-        return l('NextDay', Globalize.format(dateOnly, 'dddd'));
+
+        return l('Next') + ' ' + dateOnly.format('dddd');
```

Comment:
No, it needs to use `l('NextDay', dateOnly.format('dddd'));`

Otherwise,  for example in french, this would create a `Suivant mardi`, if you simply concatenate the strings.

The `l('NextDay', dateOnly.format('dddd'));` construct would allow us to re-order the words correctly in `strings.json`